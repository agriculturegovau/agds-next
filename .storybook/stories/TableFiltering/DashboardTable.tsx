import { VisuallyHidden } from '@ag.ds-next/react/a11y';
import { StatusBadge, StatusBadgeTone } from '@ag.ds-next/react/badge';
import { SkeletonText } from '@ag.ds-next/react/skeleton';
import {
	Table,
	TableBody,
	TableCaption,
	TableCell,
	TableHead,
	TableHeader,
	TableWrapper,
} from '@ag.ds-next/react/table';
import { TextLink } from '@ag.ds-next/react/text-link';
import { Text } from '@ag.ds-next/react/text';
import { Fragment } from 'react';
import { BusinessForAudit } from './lib/generateBusinessData';

export const DashboardTable = ({
	data,
	loading,
}: {
	data: BusinessForAudit[];
	loading?: boolean;
}) => {
	if (!loading && data.length === 0) {
		return <Text>No results</Text>;
	}

	return (
		<TableWrapper>
			<Table>
				<TableCaption>Establishments</TableCaption>
				<TableHead>
					<tr>
						<TableHeader width="30%" scope="col">
							Business name
						</TableHeader>
						<TableHeader scope="col">City</TableHeader>
						<TableHeader scope="col">State</TableHeader>
						<TableHeader textAlign="right" scope="col">
							Employees
						</TableHeader>
						<TableHeader textAlign="right" scope="col">
							Date registered
						</TableHeader>
						<TableHeader scope="col">Status</TableHeader>
					</tr>
				</TableHead>
				<TableBody>
					{loading ? (
						<Fragment>
							{Array.from(Array(10).keys()).map((i) => (
								<tr key={i}>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
									<TableCell>
										<SkeletonText />
										<VisuallyHidden>Loading</VisuallyHidden>
									</TableCell>
								</tr>
							))}
						</Fragment>
					) : (
						<Fragment>
							{data.map(
								({
									id,
									businessName,
									city,
									state,
									numberOfEmployees,
									dateRegistered,
									status,
								}) => {
									return (
										<tr key={id}>
											<TableCell>
												<TextLink href={`#${id}`}>{businessName}</TextLink>
											</TableCell>
											<TableCell>{city}</TableCell>
											<TableCell>{state}</TableCell>
											<TableCell textAlign="right">
												{numberOfEmployees}
											</TableCell>
											<TableCell textAlign="right">
												{dateRegistered.toLocaleDateString()}
											</TableCell>
											<TableCell>
												<StatusBadge {...STATUS_MAP[status]} />
											</TableCell>
										</tr>
									);
								}
							)}
						</Fragment>
					)}
				</TableBody>
			</Table>
		</TableWrapper>
	);
};

const STATUS_MAP = {
	notBooked: {
		label: 'Not booked',
		tone: 'neutral',
	},
	booked: {
		label: 'Booked',
		tone: 'info',
	},
	completed: {
		label: 'Completed',
		tone: 'success',
	},
	cancelled: {
		label: 'Cancelled',
		tone: 'error',
	},
} as const;
