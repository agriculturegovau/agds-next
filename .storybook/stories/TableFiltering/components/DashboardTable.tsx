import { VisuallyHidden } from '@ag.ds-next/react/a11y';
import { StatusBadge } from '@ag.ds-next/react/badge';
import { SkeletonBox, SkeletonText } from '@ag.ds-next/react/skeleton';
import {
	Table,
	TableBody,
	TableCaption,
	TableCell,
	TableHead,
	TableWrapper,
} from '@ag.ds-next/react/table';
import { TextLink } from '@ag.ds-next/react/text-link';
import { Text } from '@ag.ds-next/react/text';
import { forwardRef, Fragment } from 'react';
import { format } from 'date-fns';
import {
	BusinessForAudit,
	BusinessForAuditWithIndex,
} from '../lib/generateBusinessData';
import { Avatar } from '@ag.ds-next/react/avatar';
import { Flex } from '@ag.ds-next/react/box';
import { GetDataSort } from '../lib/getData';
import {
	DashboardTableHeader,
	DashboardTableHeaderBaseProps,
} from './DashboardTableHeader';

const DashboardTableRowAssignee = ({
	assignee,
}: {
	/** The name of the assignee, if set */
	assignee: string | undefined;
}) => {
	if (!assignee) {
		return <TableCell>-</TableCell>;
	}

	return (
		<TableCell>
			<Flex alignItems="center" gap={0.25}>
				<Avatar name={assignee} size="sm" />
				<Text>{assignee}</Text>
			</Flex>
		</TableCell>
	);
};

export type DashboardTableProps = {
	/** The caption for the table */
	caption: string;
	/** The data to display in the table */
	data: BusinessForAuditWithIndex[];
	/** The current sort */
	sort?: GetDataSort;
	/** A function to set the sort */
	setSort?: (sort: GetDataSort) => void;
	/** Whether the table is loading */
	loading?: boolean;
	/** The total number of items found in the search */
	totalItems?: number;
	/** The number of items to display per page */
	itemsPerPage?: number;
};

export const tableId = 'dashboard-table';

export const DashboardTable = forwardRef<HTMLTableElement, DashboardTableProps>(
	function DashboardTable(
		{
			caption,
			data,
			loading,
			sort,
			setSort,
			itemsPerPage = 10,
			totalItems,
		}: DashboardTableProps,
		ref
	) {
		const isTableSortable = !!sort || !!setSort;

		if (!loading && data.length === 0) {
			return <Text>No results</Text>;
		}

		return (
			<TableWrapper>
				<Table
					aria-rowcount={totalItems}
					id={tableId}
					ref={ref}
					tabIndex={-1}
					tableLayout="fixed"
				>
					<TableCaption>
						{caption}
						<VisuallyHidden>
							, column headers with buttons are sortable.
						</VisuallyHidden>
					</TableCaption>
					<TableHead>
						<tr>
							{headers.map(
								({
									label,
									sortKey,
									textAlign,
									width,
									isSortable: isFieldSortable,
								}) => {
									const isFieldTheActiveSortField = sort?.field === sortKey;
									const clickHandler = () =>
										setSort?.({
											field: sortKey,
											order:
												sort?.field === sortKey && sort?.order === 'ASC'
													? 'DESC'
													: 'ASC',
										});

									return (
										<DashboardTableHeader
											key={sortKey}
											textAlign={textAlign}
											width={width}
											isSortable={isTableSortable && isFieldSortable}
											sort={isFieldTheActiveSortField ? sort.order : undefined}
											onClick={clickHandler}
										>
											{label}
										</DashboardTableHeader>
									);
								}
							)}
						</tr>
					</TableHead>
					<TableBody>
						{loading ? (
							<Fragment>
								{Array.from(Array(itemsPerPage).keys()).map((i) => (
									<tr key={i}>
										<TableCell>
											<SkeletonText />
											<VisuallyHidden>Loading</VisuallyHidden>
										</TableCell>
										<TableCell>
											<SkeletonText />
											<VisuallyHidden>Loading</VisuallyHidden>
										</TableCell>
										<TableCell>
											<SkeletonText />
											<VisuallyHidden>Loading</VisuallyHidden>
										</TableCell>
										<TableCell>
											<SkeletonText />
											<VisuallyHidden>Loading</VisuallyHidden>
										</TableCell>
										<TableCell>
											<SkeletonBox height={32} />
											<VisuallyHidden>Loading</VisuallyHidden>
										</TableCell>
									</tr>
								))}
							</Fragment>
						) : (
							<Fragment>
								{data.map(
									({
										index,
										id,
										assignee,
										businessName,
										city,
										state,
										requestDate,
										status,
									}) => {
										// Adding 1 because the table header row is the first row
										const rowIndex = index + 1;
										return (
											<tr key={id} aria-rowindex={rowIndex}>
												<TableCell as="th" scope="row">
													<TextLink href={`#${id}`}>{businessName}</TextLink>
												</TableCell>
												<DashboardTableRowAssignee assignee={assignee} />
												<TableCell>
													{city}, {state}
												</TableCell>
												<TableCell textAlign="right">
													{format(requestDate, 'dd/MM/yyyy')}
												</TableCell>
												<TableCell>
													<StatusBadge {...STATUS_MAP[status]} />
												</TableCell>
											</tr>
										);
									}
								)}
							</Fragment>
						)}
					</TableBody>
				</Table>
			</TableWrapper>
		);
	}
);

const STATUS_MAP = {
	notBooked: {
		label: 'Not booked',
		tone: 'neutral',
	},
	booked: {
		label: 'Booked',
		tone: 'info',
	},
	completed: {
		label: 'Completed',
		tone: 'success',
	},
	cancelled: {
		label: 'Cancelled',
		tone: 'error',
	},
} as const;

const headers: ({
	label: string;
	sortKey: keyof BusinessForAudit;
	isSortable: boolean;
} & DashboardTableHeaderBaseProps)[] = [
	{
		label: 'Business name',
		sortKey: 'businessName',
		width: { xs: '20rem', lg: 'auto' },
		isSortable: true,
	},
	{
		label: 'Assignee',
		sortKey: 'assignee',
		width: '12rem',
		isSortable: true,
	},
	{
		label: 'City',
		sortKey: 'city',
		width: { xs: '16rem', lg: 'auto' },
		isSortable: false,
	},
	{
		label: 'Date registered',
		sortKey: 'requestDate',
		textAlign: 'right',
		width: '12rem',
		isSortable: true,
	},
	{
		label: 'Status',
		sortKey: 'status',
		width: '11rem',
		isSortable: true,
	},
];
