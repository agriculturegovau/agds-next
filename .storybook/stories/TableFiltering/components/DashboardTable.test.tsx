import '@testing-library/jest-dom';
import 'html-validate/jest';
import { cleanup, render } from '../../../../test-utils';
import { DashboardTable, DashboardTableProps } from './DashboardTable';
import { useTestData } from '../lib/testData';
import { useSortAndFilter } from '../lib/useSortAndFilter';

afterEach(cleanup);

const DashboardTableTest = ({ loading }: { loading: boolean }) => {
	const {
		sort,
		filters,
		pagination,
		resetPagination,
		setSort,
		setFilters,
		resetFilters,
		removeFilter,
		setPagination,
	} = useSortAndFilter();

	const { data, totalPages, totalItems } = useTestData({
		filters,
		pagination,
		sort,
	});

	const caption = 'Table caption';

	return (
		<DashboardTable
			{...{
				sort,
				setSort,
				filters,
				setFilters,
				resetFilters,
				removeFilter,
				pagination,
				setPagination,
				resetPagination,
				totalPages,
				totalItems,
				loading,
				data,
				caption,
			}}
		/>
	);
};

describe('DashboardTable', () => {
	describe('when loading', () => {});

	describe('when data is loaded', () => {
		it('renders correctly', () => {
			const { container } = render(<DashboardTableTest loading={false} />);
			expect(container).toMatchSnapshot();
		});

		it('renders a valid HTML structure', () => {
			const { container } = render(<DashboardTableTest loading={false} />);
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],

				rules: {
					'no-inline-style': 'off',
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
		});
	});
});
