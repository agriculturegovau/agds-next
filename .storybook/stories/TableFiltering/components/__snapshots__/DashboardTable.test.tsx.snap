// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardTable when data is loaded renders correctly 1`] = `
<div>
  <div
    class="css-2ebmvj-TableWrapper"
  >
    <table
      class="css-57emj1-boxStyles-Table"
      id="dashboard-table"
    >
      <caption
        class="css-qt4zh6-boxStyles-Text-TableCaption"
      >
        Table caption
        <span
          class="css-avudkd-VisuallyHidden"
        >
          , column headers with buttons are sortable.
        </span>
      </caption>
      <thead
        class="css-11khhn7-boxStyles"
      >
        <tr>
          <th
            aria-sort="ascending"
            class="css-xp2i77-boxStyles-DashboardTableHeader"
            scope="col"
          >
            <button
              class="css-1o4ggwj-BaseButton-boxStyles-DashboardTableHeader"
              type="button"
            >
              Business name
              <svg
                aria-hidden="true"
                class="css-dgj9fa-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 8L12 17L3 8"
                />
              </svg>
            </button>
          </th>
          <th
            class="css-83w9b1-boxStyles-DashboardTableHeader"
            scope="col"
          >
            <button
              class="css-1o4ggwj-BaseButton-boxStyles-DashboardTableHeader"
              type="button"
            >
              Assignee
              <svg
                aria-hidden="true"
                class="css-dgj9fa-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 8L11.8232 3.67678C11.9209 3.57915 12.0791 3.57915 12.1768 3.67678L16.5 8"
                />
                <path
                  d="M16.5 16L12.1768 20.3232C12.0791 20.4209 11.9209 20.4209 11.8232 20.3232L7.5 16"
                />
              </svg>
            </button>
          </th>
          <th
            class="css-6jlnw2-boxStyles-TableHeader"
            scope="col"
          >
            City
          </th>
          <th
            class="css-rq1mcj-boxStyles-DashboardTableHeader"
            scope="col"
          >
            <button
              class="css-1o4ggwj-BaseButton-boxStyles-DashboardTableHeader"
              type="button"
            >
              Employees
              <svg
                aria-hidden="true"
                class="css-dgj9fa-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 8L11.8232 3.67678C11.9209 3.57915 12.0791 3.57915 12.1768 3.67678L16.5 8"
                />
                <path
                  d="M16.5 16L12.1768 20.3232C12.0791 20.4209 11.9209 20.4209 11.8232 20.3232L7.5 16"
                />
              </svg>
            </button>
          </th>
          <th
            class="css-1uhswxm-boxStyles-DashboardTableHeader"
            scope="col"
          >
            <button
              class="css-1o4ggwj-BaseButton-boxStyles-DashboardTableHeader"
              type="button"
            >
              Date registered
              <svg
                aria-hidden="true"
                class="css-dgj9fa-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 8L11.8232 3.67678C11.9209 3.57915 12.0791 3.57915 12.1768 3.67678L16.5 8"
                />
                <path
                  d="M16.5 16L12.1768 20.3232C12.0791 20.4209 11.9209 20.4209 11.8232 20.3232L7.5 16"
                />
              </svg>
            </button>
          </th>
          <th
            class="css-83w9b1-boxStyles-DashboardTableHeader"
            scope="col"
          >
            <button
              class="css-1o4ggwj-BaseButton-boxStyles-DashboardTableHeader"
              type="button"
            >
              Status
              <svg
                aria-hidden="true"
                class="css-dgj9fa-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 8L11.8232 3.67678C11.9209 3.57915 12.0791 3.57915 12.1768 3.67678L16.5 8"
                />
                <path
                  d="M16.5 16L12.1768 20.3232C12.0791 20.4209 11.9209 20.4209 11.8232 20.3232L7.5 16"
                />
              </svg>
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        class="css-8u32cd-boxStyles"
      >
        <tr
          aria-rowindex="1"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#b2c59ecc-1146-4c4b-8114-ba5b0ca48380"
            >
              Agriculture Victoria
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                GR
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                George Russell
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Melbourne
            , 
            VIC
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            679
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            20/11/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-766z4q-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1h89p7b-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.7525 1.79686C16.5646 1.6069 16.3086 1.5 16.0414 1.5H8.04987C7.78544 1.5 7.53178 1.60473 7.34437 1.79128L1.79451 7.31592C1.60598 7.50359 1.5 7.75863 1.5 8.02464V15.8844C1.5 16.1478 1.60386 16.4005 1.78903 16.5877L7.34344 22.2032C7.53128 22.3931 7.7873 22.5 8.05441 22.5H15.9501C16.2146 22.5 16.4682 22.3953 16.6556 22.2087L22.2055 16.6841C22.394 16.4964 22.5 16.2414 22.5 15.9754V8.02003C22.5 7.75675 22.3962 7.5041 22.2111 7.31689L16.7525 1.79686Z"
                />
                <path
                  d="M8 16L16 8"
                />
                <path
                  d="M16 16L8 8"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Cancelled
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="2"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#1ca8ae5c-0616-4b2b-a821-20933fff5d73"
            >
              Auer - Keeling
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                GR
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                George Russell
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Dennisfield
            , 
            TAS
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            389
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            20/01/2023
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-1w1m57w-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1esj8rc-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10.5"
                />
                <path
                  d="M7 13L10 16L17 9"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Completed
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="3"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#29c603ba-dc62-486d-8f0c-9d45d86ab109"
            >
              Aussie Acres Produce
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                OP
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                Oscar Piastri
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            North Lauretta
            , 
            QLD
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            791
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            03/01/2023
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-766z4q-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1h89p7b-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.7525 1.79686C16.5646 1.6069 16.3086 1.5 16.0414 1.5H8.04987C7.78544 1.5 7.53178 1.60473 7.34437 1.79128L1.79451 7.31592C1.60598 7.50359 1.5 7.75863 1.5 8.02464V15.8844C1.5 16.1478 1.60386 16.4005 1.78903 16.5877L7.34344 22.2032C7.53128 22.3931 7.7873 22.5 8.05441 22.5H15.9501C16.2146 22.5 16.4682 22.3953 16.6556 22.2087L22.2055 16.6841C22.394 16.4964 22.5 16.2414 22.5 15.9754V8.02003C22.5 7.75675 22.3962 7.5041 22.2111 7.31689L16.7525 1.79686Z"
                />
                <path
                  d="M8 16L16 8"
                />
                <path
                  d="M16 16L8 8"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Cancelled
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="4"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#bd25775e-f874-45ed-9f64-4ce89a077412"
            >
              Aussie Agribusiness
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                OP
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                Oscar Piastri
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Sydney
            , 
            NSW
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            645
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            04/05/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-13gpbg6-boxStyles-StatusBadge"
            >
              <div
                class="css-1ttchuf-boxStyles-NeutralDot"
              />
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Not booked
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="5"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#a1e8f4e9-c8a3-43a8-8813-c81103dda636"
            >
              Aussie Agro Supplies
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                GR
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                George Russell
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Porterville
            , 
            TAS
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            858
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            05/08/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-766z4q-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1h89p7b-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.7525 1.79686C16.5646 1.6069 16.3086 1.5 16.0414 1.5H8.04987C7.78544 1.5 7.53178 1.60473 7.34437 1.79128L1.79451 7.31592C1.60598 7.50359 1.5 7.75863 1.5 8.02464V15.8844C1.5 16.1478 1.60386 16.4005 1.78903 16.5877L7.34344 22.2032C7.53128 22.3931 7.7873 22.5 8.05441 22.5H15.9501C16.2146 22.5 16.4682 22.3953 16.6556 22.2087L22.2055 16.6841C22.394 16.4964 22.5 16.2414 22.5 15.9754V8.02003C22.5 7.75675 22.3962 7.5041 22.2111 7.31689L16.7525 1.79686Z"
                />
                <path
                  d="M8 16L16 8"
                />
                <path
                  d="M16 16L8 8"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Cancelled
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="6"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#e5ad8c98-8434-43ca-b8e5-a0a6973c3d17"
            >
              Australian Cattle Co.
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                GR
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                George Russell
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Rohanview
            , 
            WA
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            338
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            15/05/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-1jpdeo9-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1curl5b-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="17"
                  y2="11"
                />
                <path
                  d="M12 7H12.01"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Booked
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="7"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#d2bf7062-fae9-4e54-8870-447833b8822e"
            >
              Barrows LLC
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            -
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            East Elnora
            , 
            SA
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            45
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            07/08/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-13gpbg6-boxStyles-StatusBadge"
            >
              <div
                class="css-1ttchuf-boxStyles-NeutralDot"
              />
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Not booked
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="8"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#2b8f7f06-719a-4482-804c-505056a1eec8"
            >
              Barton LLC
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            -
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Port Rosalindaberg
            , 
            NSW
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            717
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            19/07/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-1w1m57w-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1esj8rc-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10.5"
                />
                <path
                  d="M7 13L10 16L17 9"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Completed
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="9"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#e825fbba-0f0b-44ca-be06-c84532fc8000"
            >
              Beahan LLC
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-q6tqn6-boxStyles"
            >
              <div
                class="css-18okgww-boxStyles-Avatar"
              >
                LN
              </div>
              <span
                class="css-6j1v6l-boxStyles-Text"
              >
                Lando Norris
              </span>
            </div>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Angiestead
            , 
            WA
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            614
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            15/03/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-1w1m57w-boxStyles-StatusBadge"
            >
              <svg
                aria-hidden="true"
                class="css-1esj8rc-Icon"
                clip-rule="evenodd"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10.5"
                />
                <path
                  d="M7 13L10 16L17 9"
                />
              </svg>
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Completed
              </span>
            </div>
          </td>
        </tr>
        <tr
          aria-rowindex="10"
        >
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <a
              class="css-38ppen-TextLink"
              href="#0ab51c04-0f39-4671-ac3f-5f23c43b259e"
            >
              Beahan, Grady and Ratke
            </a>
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            -
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            Fort Jeremyboro
            , 
            SA
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            808
          </td>
          <td
            class="css-c8qqgx-boxStyles-TableCell"
          >
            03/09/2022
          </td>
          <td
            class="css-18k5iax-boxStyles-TableCell"
          >
            <div
              class="css-13gpbg6-boxStyles-StatusBadge"
            >
              <div
                class="css-1ttchuf-boxStyles-NeutralDot"
              />
              <span
                class="css-10dvol5-boxStyles-Text-StatusBadge"
              >
                Not booked
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
